src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -235,6 +288,15 @@ private async Task CopyToAsyncCore(Stream destination, CancellationToken cancell
                 }
             }
 
+            private bool PeekChunkFromConnectionBuffer()
+            {
+                ReadOnlyMemory<byte> bytesRead = ReadChunkFromConnectionBuffer(maxBytesToRead: 0, cancellationRegistration: default);
+
+                // ReadChunkFromConnectionBuffer will return 'default' (NullRef with Length=0) if no data is available or
+                // _connection.RemainingBuffer.Slice(0, bytesToConsume) if there is.
+                return !Unsafe.IsNullRef(ref MemoryMarshal.GetReference(bytesRead.Span));
```

Comment:
I wanted to avoid changing the parsing loop below too much here. Should we be using something like a sentinel buffer object to signal available data instead?

Reply:
My suggestion would be to use a bool to explicitly indicate EOF and return this from ReadChunkFromConnectionBuffer. That matches what we do in other cases like HTTP2 (TryReadFromBuffer) etc.